NightscoutUploadKit/NightscoutUploader.swift

```diff
@@ -483,6 +565,32 @@ public class NightscoutUploader {
         }
     }
     
+    func postToNS<T: Decodable>(_ json: Any, url: URL, jsonReturnType: T.Type?) async throws -> T {
+        let data = try await postToNS(json, url: url)
+        return try JSONDecoder().decode(T.self, from: data)
+    }
+    
```

Comment:
I include a version that just returns Data too - which can be useful if the endpoint doesn't return JSON.